packages/obonode/obojobo-chunks-figure/image-properties-modal.js

```diff
@@ -59,15 +81,46 @@ class ImageProperties extends React.Component {
 			>
 				<div className="image-properties">
 					<label htmlFor="obojobo-draft--chunks--figure--url">URL:</label>
-					<input
-						type="text"
-						id="obojobo-draft--chunks--figure--url"
-						value={this.state.url || ''}
-						onChange={this.handleURLTextChange.bind(this)}
-						ref={this.inputRef}
-						size="50"
-						placeholder="Web Address of the Image"
-					/>
+					<div className="flex-container">
+						<div>
+							<input
+								type="text"
+								id="obojobo-draft--chunks--figure--url"
+								value={this.state.urlInputText || ''}
+								onChange={this.handleURLTextChange.bind(this)}
+								ref={this.inputRef}
+								size="50"
+								placeholder="Web Address of the Image"
+							/>
+						</div>
+						<div id="flex-item-2">Or</div>
+						<div id="flex-item-3">
+							<label htmlFor="image-file-input">
+								<input
+									type="file"
+									id="image-file-input"
+									onChange={this.handleFileChange.bind(this)}
+								/>
+								<span className="upload">Upload</span>
+							</label>
+						</div>
+					</div>
+
+					<div className="flex-container" id="image-container">
+						<Image
+							chunk={{
+								modelState: {
+									url: this.state.url || 'https://via.placeholder.com/140x100?text=Your+Image+Here',
+									width: 140,
+									height: 100,
```

Comment:
Defining both the width and height causes the preview image to be distorted. It works fine when uploaded but I'm concerned authors would think they uploaded a bad image.

Prefer to set the preview image to ...*x100 and restrict the height only to 100px, letting the width be auto. Then, setting max-width: 100% on the preview img element to cover cases where someone uploads a really wide image.